src/server/game/Entities/Player/Player.cpp

```diff
@@ -1929,7 +1929,7 @@ bool Player::TeleportTo(uint32 mapid, float x, float y, float z, float orientati
 
         // this will be used instead of the current location in SaveToDB
         m_teleport_dest = WorldLocation(mapid, x, y, z, orientation);
-        SetFallInformation(0, z);
+        SetFallInformation(0, GetPositionZ());
```

Comment:
We are not yet teleported, don't see any reason why we should relocate fallZ to teleportZ. This is done when teleport ack packet is received (movement cannot be processed while being teleported)